engine/compiler/compiler.go

```diff
@@ -427,6 +437,35 @@ func (c *Compiler) Compile(ctx context.Context, args runtime.CompilerArgs) runti
 		}
 	}
 
+	// create internal steps if build running in debug mode
+	if c.Tmate.Enabled && args.Build.Debug && pipeline.Platform.OS != "windows" {
+		// first we need to add an internal setup step to the pipeline
+		// to copy over the tmate binary. Internal steps are not visible
+		// to the end user.
+		spec.Internal = append(spec.Internal, &engine.Step{
+			ID:         random(),
+			Pull:       engine.PullIfNotExists,
+			Image:      image.Expand(c.Tmate.Image),
+			Entrypoint: []string{"/bin/sh", "-c"},
+			Command:    []string{"cp /bin/tmate /usr/drone/bin/"},
+		})
+
+		// next we create a temporary volume to share the tmate binary
+		// with the pipeline containers.
+		for _, step := range append(spec.Steps, spec.Internal...) {
+			step.Volumes = append(step.Volumes, &engine.VolumeMount{
+				Name: "_addons",
+				Path: "/usr/drone/bin",
+			})
+		}
+		spec.Volumes = append(spec.Volumes, &engine.Volume{
+			EmptyDir: &engine.VolumeEmptyDir{
+				ID:   random(),
+				Name: "_addons",
```

Comment:
can you add a linting rule that prevents a user from creating a volume named `_addons`?
https://github.com/drone-runners/drone-runner-docker/blob/master/engine/linter/linter.go#L160

Reply:
Done